### PR TITLE
Context_free: add a special_function' variant

### PR DESCRIPTION
**Pull request by Octachron**
Parsing `Longindent.t`s is quite unpractical because `Longident.t` are used as an IR for different and distinct syntaxic classes. This means that interpreting the string argument of the `Context_free.special_function` is less trivial than it appears.

To circumvent this issue, this commit takes the view that in complex cases the user themself can still easily specify the identifier that they intended to use.

Thus it sounds sensible to provides a `special_function'` variant function which takes directly a `Longident.t` argument for advanced uses.

### CODE REVIEW

**Review by NathanReb — APPROVED** [review]
Looks good, thanks!

**Comment by NathanReb** [comment]
I missed that the DCO check's unhappy. If you can fix this, I'll merge right after!

**Comment by Octachron** [comment]
The identity confusion and the location of the change entry should be fixed now.

**Comment by NathanReb** [comment]
I just fixed the formatting, will merge when the CI's green.

**Pull request merged**
